nameable.rb

```diff
@@ -0,0 +1,30 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class BaseDecorator < Nameable
+  attr_accessor :component
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
```

Comment:
- [ ] **[OPTIONAL]** I would recommend you move this  class to its own file in keeping with Ruby convention